mutt2task.py

```diff
@@ -13,21 +13,38 @@
 import sys
 import email
 import re
+import errno
 
 from email.header import decode_header
 from subprocess import call, Popen, PIPE
 
+def rollback():
+    print("INFO: Rolling back incomplete task/note creation:")
+    call(['task', 'rc.confirmation=off', 'undo'])
+
 home_dir = os.path.expanduser('~')
 
-notes_dir = ""
+notes_folder = ""
 notes_folder_pat = re.compile("^[^#]*\s*NOTES_FOLDER\s*=\s*(.*)$")
 for line in open("%s/.taskopenrc" % home_dir, "r"):
     match = notes_folder_pat.match(line)
     if match:
         notes_folder = match.group(1).replace('"', '')
 
-if not notes_dir:
-    notes_dir = "%s/.tasknotes" % home_dir
+if "$HOME" in notes_folder:
+    notes_folder = notes_folder.replace("$HOME", home_dir)
+
+if not notes_folder:
+    notes_folder = "%s/.tasknotes" % home_dir
+
+try:
+    os.mkdir(notes_folder, 750)
```

Comment:
Why not to check if directory exist before do `mkdir`?

Reply:
That is why the try..except block is there. It catches exceptions like directory exists and others which we then can handle explicitely.

Reply:
Yeah, I see that. But it seems that simple if condition will work faster than throwing exception every time.

Reply:
IMO, the penalty for using try..except isn't worth the paper it is written on, also program flow control and exception handling should not be used interchangeably.

Then, please consider lines 40..47, not only the try..except construct. We have to check for existence but also have to catch general failure. That makes 2 checks. Always. IMO, in such a case a try..except is way clearer than stacked if clauses. 

In general, trying to avoid if-ladders is a good thing to do. Furthermore, consider the race condition when if (dir exists) yields false, and after the check right before the mkdir is called some other party creates the dir. mkdir will fail and raise an exception anyway. 

So, again generally speaking, it is always safer to catch exceptions where they might get thrown.

C.f. [secure programming](https://dwheeler.com/secure-programs/Secure-Programs-HOWTO/index.html), in particular regarding [python mantra EAFP](https://docs.python.org/3/glossary.html#term-eafp)